### PR TITLE
Ensure support for newer CRC (>=2.32)

### PR DESCRIPTION
**Pull request by cjeanner**
dnsmasq container was removed from newer CRC releases[1], leading to a
crash whenever we want to update its configuration.

This patch updates both reproducer and Zuul to ensure it's compatible
with older and newer releases.

[1] https://github.com/crc-org/snc/commit/35fce67a31e469f84e1047cc204771d12b5f6434

As a pull request owner and reviewers, I checked that:
- [X] Appropriate testing is done and actually running

### CODE REVIEW

**Review by d34dh0r53 — COMMENTED** [review]
/lgtm

File: ci/playbooks/multinode-customizations.yml [hunk 1]
```diff
@@ -18,6 +26,11 @@
             _decoded_net_env.crc_ci_bootstrap_networks_out.crc.default.ip | ansible.utils.ipaddr('address')
           }}
         _crc_default_net_dns: "{{ _decoded_net_env.crc_ci_bootstrap_provider_dns | default([]) }}"
+        _dnsmasq_config: >-
+          {{
+            _dnsmasq.stat.exists |
+            ternary('/srv/dnsmasq.conf', '/etc/dnsmasq.d/crc-dnsmasq.conf')
```
Comment (marios) [inline ci/playbooks/multinode-customizations.yml:32]: maybe can be
```
ternary(_dnsmasq.stat.path, '/etc/dnsmasq.d/crc-dnsmasq.conf')
```
to prevent duplication

**Review by pablintino — CHANGES_REQUESTED** [review]
This requires deeper analysis. Why are we still calling the crc-pre? And even worst, why is the reproducer not failing if crc-pre is a service that exists only in the rhos-infra crc images?

**Comment by cjeanner** [comment]
> This requires deeper analysis. Why are we still calling the crc-pre? And even worst, why is the reproducer not failing if crc-pre is a service that exists only in the rhos-infra crc images?

reproucer doesn't use crc-pre. I didn't want to dig too deep in the whole bootstrap playbook, to be fair.

File: ci/playbooks/multinode-customizations.yml (hunk 1 above)
Comment (cjeanner) [reply ci/playbooks/multinode-customizations.yml]: good point

File: ci/playbooks/multinode-customizations.yml [hunk 2]
```diff
@@ -9,6 +9,14 @@
       ansible.builtin.slurp:
         src: "/etc/ci/env/networking-info.yml"
 
+    - name: Check which dnsmask config we must edit
```
[the remark this answers is not in the document]
Comment (cjeanner) [reply ci/playbooks/multinode-customizations.yml]: corrected

**Comment by cjeanner** [comment]
@pablintino I think we can re-open that PR for review and unhold it? Once we get the new nested image ready, molecule should work properly.
The `multinode-customize.yml` play is now converged with what the reproducer `configure_crc.yml` tasks file does, ensuring we're not relying on the custom `pre-crc` service anymore.
The nmcli changes in that play are actually testing in any CI run we have here - the only thing "not tested in CI" is the dnsmasq service reload - I don't expect it to fail at that point, to be fair... ?

**Comment by cjeanner** [comment]
Molecule's failing due to nested virt (for a change), there's an on-going issue with that in RDO CI.

IMHO, the change is safe to merge, up to the team.

File: ci/playbooks/multinode-customizations.yml [hunk 3]
```diff
@@ -9,85 +9,135 @@
...
-        state: absent
+            # TODO: ensure we know CRC IP on osp_trunk interface
+            # Needs the networking_mapper output.
+            - name: Configure local DNS for CRC pod
+              become: true
+              register: last_modification
+              ansible.builtin.replace:
+                path: "{{ _dnsmasq_config }}"
+                regexp: "192.168.130.11"
+                replace: "{{ _crc_default_net_ip }}"
```
Comment (marios) [inline ci/playbooks/multinode-customizations.yml:102]: hard to follow the indentation here but... are we really still inside the outer block here so we have this available default_net_ip (line 27)

File: ci/playbooks/multinode-customizations.yml [hunk 4]
```diff
@@ -9,85 +9,135 @@
...
+              ansible.builtin.service:
+                name: NetworkManager
+                state: reloaded
+
+        - name: Configure dnsmasq
+          vars:
+            _dnsmasq_config: >-
+              {{
+                _dnsmasq.stat.exists |
+                ternary('/srv/dnsmasq.conf', '/etc/dnsmasq.d/crc-dnsmasq.conf')
```
Comment (marios) [inline ci/playbooks/multinode-customizations.yml:77]: added an earlier comment perhaps use _dnsmasq.stat.path as the true result instead of repeating the path.

do we *also* want to stat /etc/dnsmasq.d/crc-dnsmasq.conf as an extra sanity then just error if neither?

**Review by marios — COMMENTED** [review]
stared again and it grew since the last visit :)

generally lgtm some inline thoughts.

holding approve for the review call

File: ci/playbooks/multinode-customizations.yml (hunk 3 above)
Comment (cjeanner) [reply ci/playbooks/multinode-customizations.yml:102]: yes, we are - there are 2 levels of blocks. I know, it may be slightly hard, but it allow a lot of code de-duplication. Also, it did properly pass CI before RDO got down - molecule tests are failing due to the nested virt issue currently roaming the jobs :(

File: ci/playbooks/multinode-customizations.yml (hunk 4 above)
Comment (cjeanner) [reply ci/playbooks/multinode-customizations.yml:77]: ah, right for the .path - missed that one while doing the deeper update.

As for checking the other file: it would fail if it doesn't exist anyway, because we don't instruct ansible to `create: true` the file.
AFAIK, the next move CRC *could* do is moving dnsmasq to the NetworkManager service... Not sure they want to go that way though.

File: ci/playbooks/multinode-customizations.yml [hunk 5]
```diff
@@ -9,85 +9,133 @@
...
-      containers.podman.podman_container:
-        name: crc-dnsmasq
-        state: absent
+            - name: Configure local DNS for CRC pod
+              become: true
+              register: last_modification
+              ansible.builtin.replace:
+                path: "{{ _dnsmasq_config }}"
+                regexp: "192.168.130.11"
+                replace: "{{ _crc_default_net_ip }}"
```
Comment (pablintino) [inline ci/playbooks/multinode-customizations.yml:100]: There is a small change in the behavior here. The old code set there the IP only, without the network prefix length, as the `_decoded_net_env.crc_ci_bootstrap_networks_out.crc.default.ip` value was filtered with `ansible.utils.ipaddr('address')`.

Comment (cjeanner) [reply ci/playbooks/multinode-customizations.yml:100]: oh, good catch. It seems to be working though, but indeed, for the sake of consistency - lemme update it.

**Review by raukadah — APPROVED** [review]

**Review by marios — APPROVED** [review]
/lgtm
